tests/tx/test_verification.py

```diff
@@ -0,0 +1,448 @@
+        _input.data = P2PKH.create_input_data(public_bytes, signature)
+
+        return tx
+
+    def _get_valid_token_creation_tx(self) -> TokenCreationTransaction:
+        add_blocks_unlock_reward(self.manager)
+        assert self.manager.wallet
+        return create_tokens(self.manager, self.manager.wallet.get_unused_address())
+
+    def test_block_verify_basic(self) -> None:
+        block = self._get_valid_block()
+
+        verify_weight_wrapped = Mock(wraps=block.verify_weight)
+        verify_reward_wrapped = Mock(wraps=block.verify_reward)
+
+        with (
+            patch.object(Block, 'verify_weight', verify_weight_wrapped),
+            patch.object(Block, 'verify_reward', verify_reward_wrapped),
+        ):
+            self.manager.verification_service.verify_basic(block)
+
+        # Block methods
+        verify_weight_wrapped.assert_called_once()
+        verify_reward_wrapped.assert_called_once()
+
+    def test_block_verify_without_storage(self) -> None:
+        block = self._get_valid_block()
+
+        verify_pow_wrapped = Mock(wraps=block.verify_pow)
+        verify_no_inputs_wrapped = Mock(wraps=block.verify_no_inputs)
+        verify_outputs_wrapped = Mock(wraps=block.verify_outputs)
+        verify_data_wrapped = Mock(wraps=block.verify_data)
+        verify_sigops_output_wrapped = Mock(wraps=block.verify_sigops_output)
+
+        with (
+            patch.object(Block, 'verify_pow', verify_pow_wrapped),
+            patch.object(Block, 'verify_no_inputs', verify_no_inputs_wrapped),
+            patch.object(Block, 'verify_outputs', verify_outputs_wrapped),
+            patch.object(Block, 'verify_data', verify_data_wrapped),
+            patch.object(Block, 'verify_sigops_output', verify_sigops_output_wrapped),
+        ):
+            block.verify_without_storage()
+
+        # Block methods
+        verify_pow_wrapped.assert_called_once()
+        verify_no_inputs_wrapped.assert_called_once()
+        verify_outputs_wrapped.assert_called_once()
+        verify_data_wrapped.assert_called_once()
+        verify_sigops_output_wrapped.assert_called_once()
+
+    def test_block_verify(self) -> None:
+        block = self._get_valid_block()
+
+        verify_pow_wrapped = Mock(wraps=block.verify_pow)
+        verify_no_inputs_wrapped = Mock(wraps=block.verify_no_inputs)
+        verify_outputs_wrapped = Mock(wraps=block.verify_outputs)
+        verify_data_wrapped = Mock(wraps=block.verify_data)
+        verify_sigops_output_wrapped = Mock(wraps=block.verify_sigops_output)
+        verify_parents_wrapped = Mock(wraps=block.verify_parents)
+        verify_height_wrapped = Mock(wraps=block.verify_height)
+
+        with (
+            patch.object(Block, 'verify_pow', verify_pow_wrapped),
+            patch.object(Block, 'verify_no_inputs', verify_no_inputs_wrapped),
+            patch.object(Block, 'verify_outputs', verify_outputs_wrapped),
```

Comment:
It's missing a wrapped call to the `verify_number_of_outputs()` which is usually called by `verify_outputs()`.

Reply:
Thanks!